src/shell/shell_cmds.cpp

```diff
@@ -608,6 +614,61 @@ static std::vector<int> calc_column_widths(const std::vector<int> &word_widths,
 	return col_widths;
 }
 
+char buffer[15] = {0};
+char *format_date(uint16_t year, uint8_t month, uint8_t day)
+{
+	char formatstring[6];
+	char date_format = dos.tables.country[0];
+	char date_separator = dos.tables.country[11];
+	sprintf(formatstring,
+	        date_format == 1 ? "D%cM%cY"
+	                         : (date_format == 2 ? "Y%cM%cD" : "M%cD%cY"),
+	        date_separator, date_separator);
+	Bitu bufferptr = 0;
+	for (Bitu i = 0; i < 5; i++) {
+		if (i == 1 || i == 3) {
+			buffer[bufferptr] = formatstring[i];
+			bufferptr++;
+		} else {
+			if (formatstring[i] == 'M')
+				bufferptr += (Bitu)sprintf(buffer + bufferptr,
+				                           "%02u", month);
+			if (formatstring[i] == 'D')
+				bufferptr += (Bitu)sprintf(buffer + bufferptr,
+				                           "%02u", day);
+			if (formatstring[i] == 'Y')
+				bufferptr += (Bitu)sprintf(buffer + bufferptr,
+				                           "%04u", year);
+		}
+	}
+	return buffer;
+}
+
+char *format_time(Bitu hour, Bitu min, Bitu sec, Bitu msec, bool full = false)
```

Comment:
Hopeful we don't need 64-bit sizes for H, M, S.
Also `const` when possible.

Reply:
Done.